src/mono/mono/mini/monovm.c

```diff
@@ -275,41 +270,45 @@ monovm_runtimeconfig_initialize (MonovmRuntimeConfigArguments *arg, MonovmRuntim
 }
 
 void
-mono_extract_and_register_properties (const char *buffer, long fileSize)
+mono_extract_and_register_properties (const char *buffer, guint64 file_size)
```

Comment:
Again I would integrate this with `mono_runtime_install_appctx_properties`. Store away the `MonovmRuntimeConfigArguments`, the `cleanup_fn` and its `arg`, and use the in the _install function.  You can read off the count of properties, then allocate the total utf16 arrays, then populate it with the monovm properties, then read from the buffer and convert to utf16 and store directly into the final array and avoid a few more copies.

Reply:
Moved.